src/components/vm/overview/cpuModal.tsx

```diff
@@ -231,10 +244,10 @@ export const CPUModal = ({ vm, maxVcpu, models }) => {
                        }>
                 <NumberInput
                     id="machines-vcpu-max-field"
-                    value={max}
-                    onMinus={() => onMaxChange(max - 1)}
-                    onPlus={() => onMaxChange(max + 1)}
-                    onChange={event => onMaxChange(event.target.value)}
+                    value={max === 0 ? "" : max}
+                    onMinus={() => onMaxChange(String(max - 1))}
+                    onPlus={() => onMaxChange(String(max + 1))}
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2185-af23a7d6-20250716-112918-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)